### ISSUE TITLE
 threads

### ISSUE THREAD

**Issue by hysong0921** [SwiftSeal/resistify#37]
How can I allocate computing resources with the latest version 1.0? I only want to assign 20 threads, but it seems that the -t parameter is no longer working.

**Comment by SwiftSeal** [comment]
Hi @hysong0921 

I removed that option in v0.6 as numpy will use all threads regardless, so it was a bit misleading. resistify should identify all available cpus and split the jackhmmer process automatically. Is it using all available cpus?

**Comment by hysong0921** [comment]
Yes, it automatically takes up the idle server resources, but I want to leave some resources available for other researchers to run their programs. If it takes up everything, others won't be able to continue their work. Is it possible for Resistify to dynamically adjust its resource usage when others submit tasks? Thank you!

**Comment by SwiftSeal** [comment]
Ahh I see! I assumed most users would be using SLURM etc so it would be handled automatically. I'll add the option back in and see if I can make it restrict the numpy process as well so it is more "true". In the meantime I'd suggest:

* Setting `--chunksize` to a very high number - more than the total number of NLRs. This will essentially make it run on two threads (but not numpy).
* If you have access to docker/apptainer, I believe you can restrict cpus via the container manager

I'll let you know when the feature is available again :)

**Comment by SwiftSeal** [comment]
Actually I think I fixed it! I won't have time to test it this year, but if you try:
```
conda install resistify
pip install git+https://github.com/swiftseal/resistify.git@threads
```
It will install a patched version which has the `--threads` argument, which numpy should now obey.

**Comment by hysong0921** [comment]
Could you please provide an example? Submitting the command like this after updating the patch doesn't seem to work:
resistify nlr --threads 10 <input.fasta>
Usage: resistify nlr [-h] [--retain] [--coconat] [--batch BATCH] [-o OUTDIR] [--models-path MODELS_PATH] [--lrr_gap LRR_GAP] [--lrr_length LRR_LENGTH] [--duplicate_gap DUPLICATE_GAP]
                     [--chunksize CHUNKSIZE]
                     input
resistify nlr: error: argument input: Input file 10 does not exist

**Comment by SwiftSeal** [comment]
Is it reporting that it is version v1.0.1? Otherwise the patch won't have installed. I'll release the update to conda in the meantime which should be easier to install

**Comment by SwiftSeal** [comment]
Okay I've now released version v1.0.1 which has this option. It should be available on conda soon :)

**Comment by SwiftSeal** [comment]
Marking as closed as option has been reenabled!
